### PR TITLE
Backend Team Delete

### PR DESCRIPTION
**Pull request by aksh0001**
# Description 

We have middleware for triggering dependent document creation for team registrations. This PR implements a dependent document deletion middleware/hook/trigger to delete associated documents when a team has been deleted.

This PR is a result of comments left in https://github.com/Researchify/Researchify/pull/198.

Note: I've also changed up the positioning of the controller and routes to follow CRUD, so we define create endpoints first, followed by read, update, and then delete (the additions are few, but the diff count is bloated because of this) to make it easier to follow. The only additions are the `deleteTeam()` controller + route, and changes in the `Team` model.

# Type of change 
- [ ] Bug fix
- [x] New feature
- [ ] Refactoring
- [ ] Documentation 

# Problem
There is no way to delete a team in the backend.

# Solution description
Implement a delete method for the backend, as well as a post hook that will delete any associated documents.

# Tests 
All unit tests pass? Y

Builds successfully? Y

Attach screenshot of the expected result (e.g. UI of the website/ postman api call result/ database changes)
Screenshot from 2021-10-04 00-07-53

Returns `204` (no content)

Provide steps in detail for reviewers to reproduce the expected result of the code changes.
Please manually verify that the dependent documents have been deleted in the database.

# Risk
Low, additive.

### CODE REVIEW

File: api/src/controllers/team.js [hunk 1]
```diff
@@ -183,6 +172,78 @@ function updateTeamMember(req, res, next) {
...
+    // validate the handle by getting user id
+    if (!process.env.TWITTER_BEARER_TOKEN) {
+      return next(
+        fillErrorObject(500, 'Missing environment variable', [
+          'No Twitter API Bearer Token found in .env file',
+        ]),
+      );
+    }
+    const response = await axios.get(
+      `https://api.twitter.com/2/users/by/username/${handle}`,
```
Comment (cyr17) [inline api/src/controllers/team.js:222]: Out of curiosity, Is octokit just for raw gitHub requests or can it be used here too?

Comment (aksh0001) [reply api/src/controllers/team.js:222]: Octokit is the official GitHub client library. But good spot, we should upgrade this code to use the official Twitter library in the future.

I haven’t used it before in NodeJS but this seems to be the one https://www.npmjs.com/package/twitter-api-v2

**Review by jfan0010 — APPROVED** [review]
lgtm

**Review by cyr17 — APPROVED** [review]
Allg , might need to fix up eslint

**Review by pmyip3 — APPROVED** [review]

**Review by rpra0006 — APPROVED** [review]

**Pull request merged**